### DISCUSSION TITLE
Force recoonect and invoke some code on each reconnection

### DISCUSSION THREAD

**Discussion by alteest** [ag2ai/faststream#1956] [General]
Hi,
I need to implement next features in application which is use faststream (RabbitMQ):
1. I need in some case force reconnection. How to do it?
2. On each connection I need to execute some code, let say I need to send something to RabbitMQ server.
I don't something like 'on_connect' hooks, etc.
I know how to do it without faststream ,but is it possible with faststream?

**Comment by Lancetnik** [comment]
Seems like a copy of #1545 + #1958
I'll check, what we can do with this

**Reply by alteest** [reply]
Not sure if it's a good solution to call some code on each reconnection. I see that callback executed on reconnection, but not sure if it will work on all cases.
```python
async def on_broker_re_connect(*args, **kwargs):
    ...

@app.after_startup
async def process_after_startup(*args, **kwargs):
    broker._connection.reconnect_callbacks.add(on_broker_re_connect)
    await on_broker_re_connect()
```

**Reply by Lancetnik** [reply]
It is access to **aio-pika** lib and should works, but I didn't check such behavior
